### PR TITLE
Unify nanomaterials and nanoelements

### PR DESCRIPTION
**Pull request by scruti**
[Jira ticket](https://regulatorydelivery.atlassian.net/browse/COSBETA-1594)
Merging the NanoElement attributes into NanoMaterial and using NanoMaterial instead of NanoElement across the service.

## Description

NanoMaterials at the moment is an empty shell, with only 2 fields that are not used anymore since they are currently being recorded at the Component (exposure routes and exposure condition).

All the valuable information about the NanoMaterial is recorded in NanoElement instances. Having 1 to many relationship between the NanoMaterial and NanoElements, but effectively having a single NanoElement per NanoMaterial and acting like a 1-to-1.

This PR resolves this data modeling issue by:
- Adding all the NanoElement DB fields/attributes into NanoMaterial.
- Bringing the NanoElement class logic into NanoMaterial, deleting some unused code.
- Adding a task that clones all the NanoElement data into their associated NanoMaterial.
- Changing all the service logic to stop using NanoElement and use NanoMaterial instead.

## Review apps

https://cosmetics-pr-2599-submit-web.london.cloudapps.digital/
https://cosmetics-pr-2599-search-web.london.cloudapps.digital/

### CODE REVIEW

File: cosmetics-web/app/models/nano_element.rb [hunk 1]
```diff
@@ -23,12 +23,6 @@ def display_name
       .reject(&:blank?).join(", ")
   end
 
-  def required?
-    purposes.blank? ||
-      (non_standard? && toxicology_required_or_empty?) ||
-      (standard? && restrictions_confirmed_required?)
-  end
-
```
Comment (scruti) [inline cosmetics-web/app/models/nano_element.rb:31]: Unused code.

File: cosmetics-web/app/models/nano_element.rb [hunk 2]
```diff
@@ -64,19 +58,6 @@ def conforms_to_restrictions?
...
-      (
-        (confirm_restrictions == "yes" && usage_confirmed_required?) ||
-       (confirm_usage == "no" && toxicology_required_or_empty?)
-      )
-  end
-
-  def usage_confirmed_required?
-    confirm_usage.nil?
-  end
-
```
Comment (scruti) [inline cosmetics-web/app/models/nano_element.rb:79]: unused code

File: cosmetics-web/app/controllers/responsible_persons/notifications/nanomaterials/build_controller.rb [hunk 3]
```diff
@@ -156,18 +157,12 @@ def update_non_standard_nanomaterial_step
...
       rerender_current_step
     end
   end
 
-  def get_next_nano_element
-    @nano_element.nano_material.nano_elements.order(:id).each_cons(2) do |element, next_element|
-      return next_element if element == @nano_element
-    end
-  end
-
```
Comment (scruti) [inline cosmetics-web/app/controllers/responsible_persons/notifications/nanomaterials/build_controller.rb:169]: Uncalled code

File: cosmetics-web/app/models/nano_element.rb (hunk 1 above)
Comment (scruti) [inline cosmetics-web/app/models/nano_element.rb:31]: Uncalled code

**Review by slawosz — APPROVED** [review]

**Pull request merged**